apps/spruce/src/pages/waterfall/BuildRow.tsx

```diff
@@ -110,38 +136,67 @@ const BuildGrid: React.FC<{
   build: Build;
   handleTaskClick: (s: string) => () => void;
   isRightmostBuild: boolean;
-}> = ({ build, handleTaskClick, isRightmostBuild }) => (
-  <BuildContainer
-    onClick={(event: React.MouseEvent) => {
-      handleTaskClick(
-        (event.target as HTMLDivElement)?.getAttribute("status") ?? "",
-      );
-    }}
-  >
-    {build.tasks.map(
-      ({ displayName, displayStatusCache, execution, id, status }) => {
-        // Use status as backup for tasks created before displayStatusCache was introduced
-        const taskStatus = (displayStatusCache || status) as TaskStatus;
-        return (
-          <SquareMemo
-            key={id}
-            data-tooltip={`${displayName} - ${taskStatusToCopy[taskStatus]}`}
-            isRightmostBuild={isRightmostBuild}
-            status={taskStatus}
-            to={getTaskRoute(id, { execution })}
-          />
+}> = ({ build, handleTaskClick, isRightmostBuild }) => {
+  const rowRef = useRef<HTMLDivElement>(null);
+  const { width } = useDimensions<HTMLDivElement>(rowRef);
+
+  const { columnWidth, setColumnWidth } = useBuildVariantContext();
+  useEffect(() => {
+    if (width !== 0 && columnWidth !== width) {
+      setColumnWidth(width);
+    }
+  }, [setColumnWidth, columnWidth, width]);
```

Comment:
Might we include this `useEffect` within `useBuildVariantContext`? It seems fairly self-contained

Reply:
i couldnt figure out how to get `width` in the context! 😖 let me know if im missing something 🥺